uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/LeaveRequestTest.php

```diff
@@ -1112,12 +1112,98 @@ public function testFindOverlappingLeaveRequestsForOneOverlappingLeaveRequest()
     $startDate = '2016-11-01';
     $endDate = '2016-11-03';
 
-    $overlappingRequests = LeaveRequest::findOverlappingLeaveRequests($contactID, $startDate, $endDate);
+    $overlappingRequests = LeaveRequest::findOverlappingLeaveRequests($contactID, $startDate, $endDate, $this->absenceType->id, LeaveRequest::REQUEST_TYPE_LEAVE);
     $this->assertCount(1, $overlappingRequests);
     $this->assertInstanceOf(LeaveRequest::class, $overlappingRequests[0]);
     $this->assertEquals($leaveRequest1->id, $overlappingRequests[0]->id);
   }
 
+  public function testFindOverlappingLeaveRequestsForHoursUnitRequests() {
+    $contactID = 1;
+
+    $testFromDate = '2018-04-13 12:00:00';
+    $testToDate = '2018-04-15 15:00:00';
+
+    $absenceTypes = [
+      'days' => AbsenceTypeFabricator::fabricate([ 'calculation_unit' => 1 ]),
+      'hours' => AbsenceTypeFabricator::fabricate([ 'calculation_unit' => 2 ])
+    ];
+
+    // Test suites: date from, date to, unit, should overlap or not
+    $overlappingRequestsTestSuites = [
+      ['2018-04-11 00:00:00', '2018-04-12 00:00:00', 'days', FALSE],
+      ['2018-04-12 00:00:00', '2018-04-13 00:00:00', 'days', TRUE],
+      ['2018-04-14 00:00:00', '2018-04-14 23:59:00', 'days', TRUE],
+      ['2018-04-15 23:59:00', '2018-04-16 00:00:00', 'days', TRUE],
+      ['2018-04-16 00:00:00', '2018-04-17 00:00:00', 'days', FALSE],
+      ['2018-04-12 00:00:00', '2018-04-13 12:00:00', 'hours', FALSE],
+      ['2018-04-12 00:00:00', '2018-04-13 14:00:00', 'hours', TRUE],
+      ['2018-04-12 00:00:00', '2018-04-13 15:00:00', 'hours', TRUE],
+      ['2018-04-14 00:00:00', '2018-04-15 15:00:00', 'hours', TRUE],
+      ['2018-04-15 15:00:00', '2018-04-16 00:00:00', 'hours', FALSE]
+    ];
+
+    foreach ($overlappingRequestsTestSuites as $overlappingRequestsTestSuite) {
+      $requestFromDate = new DateTime($overlappingRequestsTestSuite[0]);
+      $requestToDate = new DateTime($overlappingRequestsTestSuite[1]);
+      $requestUnit = $overlappingRequestsTestSuite[2];
+      $shouldRequestOverlap = $overlappingRequestsTestSuite[3];
+      $absenceType = $absenceTypes[$requestUnit];
+
+      $leaveRequestToTest = LeaveRequestFabricator::fabricateWithoutValidation([
+        'type_id' => $absenceType->id,
+        'contact_id' => $contactID,
+        'from_date' => $requestFromDate->format('YmdHis'),
+        'to_date' => $requestToDate->format('YmdHis'),
+      ], true);
```

Comment:
No need to pass TRUE to the fabricator, this will create balance changes which are not necessary for the scope of this test.

Reply:
Done.